plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+            # This demonstrates that inline definitions are still supported if
+            # 'DeploymentTemplate' is omitted.
+            - name: "custom_tg"
+              partitioning:
+                shards: 1
+              shardTemplate:
+                pools:
+                  - type: "replica"
+                    cell: "us-west-2"
+                    replicas: 2 
+                    dataVolumeClaimTemplate:
+                      accessModes: ["ReadWriteOnce"]
+                      resources:
+                        requests:
+                          storage: "75Gi"
+                    postgres:
+                      resources:
+                        requests:
+                          cpu: "1"
+                          memory: "1Gi"
+                        limits:
+                          cpu: "1"
+                          memory: "2Gi"
+                    multipooler:
+                      resources:
+                        requests:
+                          cpu: "100m"
+                          memory: "128Mi"
+                        limits:
+                          cpu: "200m"
+                          memory: "256Mi"
+
+# --- Status ---
+status:
+  observedGeneration: 1
+  globalTopoServer:
+    etcd:
+      available: "True"
+  conditions:
+    - type: Available
+      status: "True"
+      lastTransitionTime: "2025-10-08T12:00:00Z"
+      message: "All components are available."
+    - type: Progressing
+      status: "False"
+      lastTransitionTime: "2025-10-08T12:00:00Z"
+      message: "Cluster reconciliation complete."
+  cells:
+    us-east-1:
+      gatewayAvailable: "True"
+      multiorchAvailable: "True" 
+      topoServerAvailable: "True" # Assuming global is available (default)
+    us-west-2:
+      gatewayAvailable: "True"
+      multiorchAvailable: "True" 
+      topoServerAvailable: "True" # Assuming managed becomes available
+  databases:
+    production_db:
+      desiredInstances: 14 
+      readyInstances: 14
+      servingWrites: "True"
+  multiadmin:
+    available: "True"
+    serviceName: "example-multigres-cluster-multiadmin"
+```
+
+
+## DeploymentTemplate CR 
+
+```yaml
+# This defines a reusable template named "standard-ha".
+# It contains specifications for multiple component types.
+# This CR won't be a child of any other resource, but used for configuration.
+# These templates are watched and reconciled by the MultigresCluster controller and ONLY when they are referenced.
+# The templates get resolved into the child CRs of MultigresCluster, the child CRs are not aware of the existence of these templates.
+# The MultigresCluster controller updates only the relevant child resource when the template is changed.
+# A deletion of a template does not trigger the deletion of the underlying configuration, but it will trigger a warning/error.
+# Or we could prevent the deletion of the template altogether while in use --> better perhaps.
```

Comment:
I think this should be more strongly worded to say "actively used DeploymentTemplate will not be allowed to delete, and this is ensured by validating webhook", or something similar.